source/Halibut/Transport/Protocol/MessageExchangeStreamTimeout.cs

```diff
@@ -0,0 +1,10 @@
+using System;
+
+namespace Halibut.Transport.Protocol
+{
+    public enum MessageExchangeStreamTimeout
```

Comment:
💭 Just a thought, but I wonder if an alternative to this enum is to acknowledge that the send/receive timeouts are a concept, and to make a class instead (which has a send timeout property and a receive timeout property), and replace the 4 `HalibutTimeoutsAndLimits` properties (TcpClientSendTimeout, TcpClientReceiveTimeout, TcpClientHeartbeatSendTimeout, TcpClientHeartbeatReceiveTimeout) with just 2 properties using this new class.

With this remodelling, we would not need this enum, and could simply pass that object into the extension methods below.

Anyway, not vital for this PR, just thought I'd float the idea 😁

Reply:
Yeh that would be a lot cleaner 👍 I'll rework

Reply:
I've reworked the change, thoughts @sburmanoctopus

Reply:
Nice one! LGTM 😁